src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePumpSettings.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Threading.Tasks;
+using Arcus.Security.Core;
+using GuardNet;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Arcus.Messaging.Pumps.ServiceBus
+{
+    public class AzureServiceBusMessagePumpSettings
+    {
+        private readonly Func<ISecretProvider, Task<string>> _getConnectionStringFromSecretFunc;
+        private readonly Func<IConfiguration, string> _getConnectionStringFromConfigurationFunc;
+        private readonly IServiceProvider _serviceProvider;
+
+        /// <summary>
+        ///     Constructor
+        /// </summary>
+        /// <param name="entityName">Name of the entity to process</param>
+        /// <param name="subscriptionName">Name of the subscription to process</param>
+        /// <param name="getConnectionStringFromConfigurationFunc">Function to look up the connection string from the configuration</param>
+        /// <param name="getConnectionStringFromSecretFunc">Function to look up the connection string from the secret store</param>
+        /// <param name="options">Options that influence the behavior of the message pump</param>
+        /// <param name="serviceProvider">Collection of services to use</param>
+        public AzureServiceBusMessagePumpSettings(string entityName, string subscriptionName,
```

Comment:
If only one of the two is used and are requred, shouldn't we then provide a ctor for each? I think that would make it more natural to use.

Reply:
We could  do that, but then we have to duplicate AddServiceBusMessagePump in our fluent registration as one would have an entity path and the other not.

Scenarios are:
1. No entity path as it's in the connection string
1. No entity path as it's in the connection string, but a subscription name
1. Entity path, but no subscription
1. Entity path with a subscription

We could change the settings for this but then we'll have to make the registration in `IServiceCollectionExtensions` more complex and I'm not sure if that's worth it.

Reply:
Ah, ok; nevermind, will indeed probably not worth the change.